athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/serde/v2/ConstraintsSerDe.java

```diff
@@ -86,7 +90,10 @@ public Constraints doDeserialize(JsonParser jparser, DeserializationContext ctxt
                 summaryMap.put(column, valueSetDeserializer.deserialize(jparser, ctxt));
             }
 
-            return new Constraints(summaryMap.build());
+            return new Constraints(summaryMap.build(),
+                    Collections.emptyList(),
+                    Collections.emptyList(),
+                    NO_LIMIT_VALUE);
```

Comment:
When the engine is upgraded to use the new version of the sdk with this new v2, isn't this new v2 going to break already deployed connectors on an older v2 version? Our "serdes" are "strict" and will throw an exception on extra fields that it doesn't recognize, see: `src/main/java/com/amazonaws/athena/connector/lambda/serde/BaseDeserializer.java:assertFieldName`

Meaning, shouldn't this be v4 instead of v2?

Reply:
No. The only reason why this would ever hit this codepath is if an engine is using an older SDK and the customer is using a newer one. In this case, the engine serializes with V2 and the SerDe would deserialize with V2 as well. Since the connector is updated, the consuming methods will expect that the new fields are initialized. 

We've already done compatibility testing for the following cases:
1. Engine has old version, connector has new version
2. Engine has new version, connector has old version
3. Engine has new version and connector has new version

Reply:
Right now the SerDe version is already V2 for the connectors before any updates, and if the engine is upgraded with a newer version of the Federation, which has a "new" v2 definition for Constraints with new fields, when the engine, even though it negotiates v2 with the connector, will end up sending new fields from the "new" v2 Constraints, which should cause the serde to throw an exception.

If you're saying that you tested it and its fine, then I'll take your word for it, but I don't see how this works given that you're changing the existing definition of v2 when there are deployments of connectors that are already on an older version of v2.

Reply:
So the reason why this works is because we're not updating the doSerialize() function with any new parameters. When the updated engine sees that the customer is using the V2 Serde, it will use the V2 object mapper and only serializes the SUMMARY object.

Reply:
You're right, I was assuming that `doSerialize` was automatically serializing all the new additional fields that were added to `Constraints` since they were annotated with `JsonProperty`, but it looks like we are manually serializing each field here ourselves.